packages/data-context/src/DataContext.ts

```diff
@@ -386,14 +377,8 @@ export class DataContext {
       }
 
       this.update((d) => {
-        if (d.currentProjectData?.testingTypeData?.activeAppData) {
```

Comment:
This is so overly complex - it's not really obvious if there's supposed to be some specific pattern here, but I collapsed everything into a single `diagnostics` property with `error: ErrorWrapper | null, warnings: ErrorWrapper[] }` and everything seems to work just fine, so I guess it's okay? 

I think the original idea was to have different ways to handle/manage errors and warnings depending on where you are in the lifecycle, eg:

- `activeAppData` - app is open
- `testingTypeData` - launchpad, after choosing a testing type
- `currentProjectData` - launchpad, a project is chosen but no testing type selected

But there isn't any obvious value in doing this, from what I can see, other than more overhead and another undocumented pattern you need to remember. For that reason I'm thinking we just delete it, and reintroduce if/when it makes sense to do so.

Reply:
+1 love this clean up!